lib/InitAll.cpp

```diff
@@ -48,6 +49,7 @@ void mlir::torch::registerOptionalInputDialects(
   registry.insert<complex::ComplexDialect, linalg::LinalgDialect,
                   memref::MemRefDialect, ml_program::MLProgramDialect,
                   scf::SCFDialect, tensor::TensorDialect, tosa::TosaDialect>();
+  tensor::registerInferTypeOpInterfaceExternalModels(registry);
```

Comment:
nit: this is [an extension](https://github.com/llvm/llvm-project/blob/bba5ee47e63298d61f6ea441a140144ce370ba92/mlir/lib/Dialect/Tensor/IR/TensorInferTypeOpInterfaceImpl.cpp#L204). We should add in [InitAll.h](https://github.com/llvm/torch-mlir/blob/main/include/torch-mlir/InitAll.h)

```cpp
void registerAllExtensions(mlir::DialectRegistry &registry);
```
and define the function in this file, `InitAll.cpp`.

Then you can do

```diff
 int main(int argc, char **argv) {
+  DialectRegistry registry;
   mlir::torch::registerAllPasses();
+  mlir::torch::registerAllExtensions(registry);
```

in `torch-mlir-opt.cpp`.